x-pack/plugins/apm/public/components/shared/KueryBar/Typeahead/index.js

```diff
@@ -20,204 +21,218 @@ const KEY_CODES = {
-            style={{
-              backgroundImage: 'none'
-            }}
-            placeholder="Search transactions and errors... (E.g. transaction.duration.us > 300000 AND context.response.status_code >= 400)"
-            inputRef={node => {
-              if (node) {
-                this.inputRef = node;
-              }
-            }}
-            disabled={this.props.disabled}
-            value={this.state.value}
-            onKeyDown={this.onKeyDown}
-            onKeyUp={this.onKeyUp}
-            onChange={this.onChangeInputValue}
-            onClick={this.onClickInput}
-            autoComplete="off"
-            spellCheck={false}
-          />
-
-          {this.props.isLoading && (
-            <EuiProgress
-              size="xs"
-              color="accent"
-              position="absolute"
+          break;
+        case KEY_CODES.TAB:
+          this.setState({ isSuggestionsVisible: false });
+          break;
+      }
+    };
+
+    selectSuggestion = suggestion => {
+      const nextInputValue =
+        this.state.value.substr(0, suggestion.start) +
+        suggestion.text +
+        this.state.value.substr(suggestion.end);
+
+      this.setState({ value: nextInputValue, index: null });
+      this.props.onChange(nextInputValue, nextInputValue.length);
+    };
+
+    onClickOutside = () => {
+      this.setState({ isSuggestionsVisible: false });
+    };
+
+    onChangeInputValue = event => {
+      const { value, selectionStart } = event.target;
+      const hasValue = Boolean(value.trim());
+      this.setState({
+        value,
+        inputIsPristine: false,
+        isSuggestionsVisible: hasValue,
+        index: null
+      });
+
+      if (!hasValue) {
+        this.props.onSubmit(value);
+      }
+      this.props.onChange(value, selectionStart);
+    };
+
+    onClickInput = event => {
+      const { selectionStart } = event.target;
+      this.props.onChange(this.state.value, selectionStart);
+    };
+
+    onClickSuggestion = suggestion => {
+      this.selectSuggestion(suggestion);
+      this.inputRef.focus();
+    };
+
+    onMouseEnterSuggestion = index => {
+      this.setState({ index });
+    };
+
+    onSubmit = () => {
+      this.props.onSubmit(this.state.value);
+      this.setState({ isSuggestionsVisible: false });
+    };
+
+    render() {
+      return (
+        <ClickOutside
+          onClickOutside={this.onClickOutside}
+          style={{ position: 'relative' }}
+        >
+          <div style={{ position: 'relative' }}>
+            <EuiFieldSearch
+              fullWidth
               style={{
-                bottom: 0,
-                top: 'initial'
+                backgroundImage: 'none'
+              }}
+              placeholder={this.props.intl.formatMessage(
+                {
+                  id: 'xpack.apm.kueryBar.searchPlaceholder',
+                  defaultMessage:
+                    'Search transactions and errors... (E.g. {queryExample})'
```

Comment:
please use …(one symbol) here
```suggestion
                     'Search transactions and errors… (E.g. {queryExample})'
```